src/Credentials/CredentialProvider.php

```diff
@@ -406,6 +442,58 @@ public static function process($profile = null, $filename = null)
         };
     }
 
+    /**
+     * Assumes role for profile that includes role_arn
+     *
+     * @return callable
+     */
+    private static function loadRoleProfile($profiles, $profileName, $filename, $stsClient)
+    {
+        $roleProfile = $profiles[$profileName];
+        $externalId = isset($roleProfile['external_id']) ? $roleProfile['external_id'] : '';
+        $mfaSerial = isset($roleProfile['mfa_serial']) ? $roleProfile['mfa_serial'] : '';
+        $roleArn = isset($roleProfile['role_arn']) ? $roleProfile['role_arn'] : '';
+        $roleSessionName = isset($roleProfile['role_session_name']) ? $roleProfile['role_session_name'] : 'aws-sdk-php-' . date('Uv');
+
+        if (!isset($profiles[$profileName]['source_profile'])) {
+            return self::reject("source_profile is not set using profile " .
+                $profileName
+            );
+        }
+
+        $sourceProfileName = $roleProfile['source_profile'];
+        if (!isset($profiles[$sourceProfileName]))
+        {
+            return self::reject("source_profile " . $sourceProfileName
+                . " using profile " . $profileName . " does not exist"
+            );
+        }
+        $sourceRegion = isset($profiles[$sourceProfileName]['region']) ? $profiles[$sourceProfileName]['region'] : 'us-east-1';
+
+        if (!isset($stsClient)) {
+            $config = [
+                'preferStaticCredentials' => true
+            ];
+            $sourceCredentials = call_user_func(CredentialProvider::ini($sourceProfileName, $filename, $config))->wait();
+            $stsClient = new StsClient([
+                'profile' => $sourceProfileName,
+                'Credentials' => $sourceCredentials,
+                'region' => $sourceRegion,
+                'version' => 'latest',
```

Comment:
Should this be keyed to a specific API version, in case future versions introduce a breaking change?